### PR TITLE
8246378: [Windows] assert on MethodHandle logging code

### PR DESCRIPTION
**Pull request by coleenp**
The MethodHandles verbose logging code tries to walk windows stacks without using the StackWalk API and fails miserably.  It was the only consumer of os::current_frame() and get_sender_for_C_frame().  For Windows, the hs_err stack walking was fixed to use the StackWalk API and NMT was used to use RtlCaptureStackBackTrace.
This change removes the stub generated for getting the fp (which isn't valid anyway on windows).  I also removed it from windows_aarch64 since it wasn't actually generated.

Tested with tier1-3 for windows-x64.
Thanks,
Coleen

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux aarch64 | Linux arm | Linux ppc64le | Linux s390x | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (6/6 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) |    |     |     |     |  ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8246378](https://bugs.openjdk.java.net/browse/JDK-8246378): [Windows] assert on MethodHandle logging code

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)
 * [Vladimir Ivanov](https://openjdk.java.net/census#vlivanov) (@iwanowww - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1321/head:pull/1321`
`$ git checkout pull/1321`

### CODE REVIEW

File: src/hotspot/cpu/x86/stubGenerator_x86_64.cpp [hunk 1]
```diff
@@ -6717,7 +6717,6 @@ address generate_avx_ghash_processBlocks() {
     StubRoutines::_fence_entry                = generate_orderaccess_fence();
 
     // platform dependent
-    StubRoutines::x86::_get_previous_fp_entry = generate_get_previous_fp();
```
Comment (iklam) [inline src/hotspot/cpu/x86/stubGenerator_x86_64.cpp:6720]: The body of `generate_get_previous_fp()` can also be removed.

**Review by iklam — APPROVED** [review]
LGTM.

File: src/hotspot/cpu/x86/stubGenerator_x86_64.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/cpu/x86/stubGenerator_x86_64.cpp:6720]: Thanks for noticing.  I meant to delete it.

**Review by iklam — APPROVED** [review]

**Review by iwanowww — APPROVED** [review]
Looks good.

**Comment by coleenp** [comment]
Thank you Ioi and Vladimir.  @iwanowww I was hoping you'd see this since it's possible that this logging is/was useful to you.

**Pull request closed without merging**